controller/konnect/reconciler_upstreamref.go

```diff
@@ -106,6 +106,11 @@ func handleKongUpstreamRef[T constraints.SupportedKonnectEntityType, TEnt constr
 		return ctrl.Result{}, fmt.Errorf("failed to update status: %w", err)
 	}
 
+	res, err := RemoveOwnerRefIfSet(ctx, cl, ent, kongUpstream)
+	if err != nil || !res.IsZero() {
+		return ctrl.Result{}, err
+	}
```

Comment:
The same as mentioned in the previous comment

Reply:
```suggestion
	if err != nil || !res.IsZero() {
		return res, err
	}
```